### ISSUE TITLE
Bower asset install path changed

### ISSUE THREAD

**Issue by steffendietz** [francoispluchino/composer-asset-plugin#165]
With https://github.com/composer/composer/commit/d98b134dc3ef7d5bd4b6e37c9621a811f8f1599d there was a change to composer, which results in `bower-asset` packages previously installed __directly__ in a directory `<project_root>/assets/vendor/` (eg. `<project_root>/assets/vendor/jquery`) now beeing placed in the directory `<project_root>/assets/vendor/bower-asset` (eg. `<project_root>/assets/vendor/bower-asset/jquery`) making the _virtual_ `bower-asset` vendor name part of the install path.

This issue is only to clarify, how you plan to deal with this. Do you plan to circumvent this change in your custom installer or do you plan to leave the new behavior imposed by composer?

So we could plan if we adjust our projects for this change or leave them as is for the moment until you release a new version of your plugin.

Thank you and best regards,

Steffen

**Comment by samdark** [comment]
Quote from Composer author:

> Just for reference, the best way to proceed and be compatible with older
and future composer versions is to override both methods and have
getPackageBasePath just forward to getInstallPath. Unless your installer
needs to take care of packages with a target dir in which case you probably
want to copy getPackageBasePath from latest composer libraryinstaller as it
does the right thing plus the forward.

**Comment by samdark** [comment]
@francoispluchino are you able to work on it anytime soon?

**Comment by francoispluchino** [comment]
Quickly? no. But, I am not against a little help.

**Comment by steffendietz** [comment]
So @samdark you're taking the lead on this? - Would be appreciated! :)

**Comment by samdark** [comment]
@francoispluchino we need a decision i.e. compatibility vs new way of how composer does things. The rest is implementation which we could try doing ourselves.

**Comment by francoispluchino** [comment]
Keep the compatibility with the last `alpha11` version and the last `dev` version of Composer.

**Comment by samdark** [comment]
Umm... `alpha1` is not compatible with `dev`. That's the issue. Do you mean fixing `composer-asset-plugin` to work like before w/ `dev` version of Composer?

**Comment by francoispluchino** [comment]
Ok, keep only the compatibility with the last `dev` version of Composer. This was already the case for all previous versions of the plugin (see the [release notes](https://github.com/francoispluchino/composer-asset-plugin/releases)).

**Comment by samdark** [comment]
Then it means no changes are required at all, right?

**Comment by samdark** [comment]
The question is how composer-asset should work from now on. Should it work as before or should it create extra dir?

**Comment by steffendietz** [comment]
I would suggest, trying to keep the previos behavior. As @Seldaek suggested this could be achived in overriding the methods in the installer.

This would ensure compatibility with future composer versions as well as current bower-asset installs.

**Comment by steffendietz** [comment]
Also, since `bower` does not have the notion of a package vendor, this would also be more in line with the handling of the actual `bower`.

**Comment by samdark** [comment]
Yep. Sounds like a plan. Do you want to take care of it, @steffendietz?

**Comment by steffendietz** [comment]
Okay. I'm on it. Give me a minute. Have to set things up. :)

**Comment by francoispluchino** [comment]
The problem is present only if we use the `extra.asset-installer-paths` option. In this case, the custom path is no correct, so we have to correct this problem: we must keep the initial behavior of the plugin.

**Example:**
```json
{
    "require": {
        "bower-asset/bootstrap": "3.3.*"
    },
    "extra": {
        "asset-installer-paths": {
            "bower-asset-library": "web/assets"
        }
    }
}
```
Before, the assets were installed:
- `web/assets/jquery`
- `web/assets/bootstrap`

Now:
- `web/assets/vendor/bower-asset/jquery`
- `web/assets/vendor/bower-asset/bootstrap`

This is not the behavior expected of the option `extra.asset-installer-paths`.

The assets must always be installed here:
- `web/assets/jquery`
- `web/assets/bootstrap`

**Comment by steffendietz** [comment]
Okay. @francoispluchino do you already have the fix in mind? Or should I try to fix this?

**Comment by francoispluchino** [comment]
Not really, therefore, you have free rein.

**Comment by steffendietz** [comment]
Ok.

**Comment by mrsombre** [comment]
@samdark Any plans to remove fxp from yii2 core requirements? I have no idea why my php framework needs a library to install web assets, even i don't need em (as of i used bower).
And today i can't do a composer update, because yii needs fxp and it is broken. Oh my.

**Comment by francoispluchino** [comment]
@mrsombre Thanks for your relevant comment! We are working to quickly resolve the change added in Composer.

But you can do a rollback of your composer version, and your problem will disappear...

**Comment by samdark** [comment]
@mrsombre it's composer that's broken, not asset plugin.

**Comment by steffendietz** [comment]
Ok. I pushed a fix in #166. Some builds are failing due to deprecation warnings. (I don't think I caused these.)

If someone could double check what I've done that would be great. :)

**Comment by nunomaduro** [comment]
@steffendietz When will this fix going be available in the 1.1.x-dev version?

**Comment by francoispluchino** [comment]
@nunomaduro Within an hour or less.

**Comment by francoispluchino** [comment]
Fixed by #166.
